script/overlayfs_chaindata.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# A tool to create/clean up the overlayfs over the chaindata directory.
+# overlayfs can be used to quickly rollback the chaindata directory
+# after testing.
+
+unset DATADIR
+unset CLEANUP
+
+PROGNAME=$(basename "$0")
+USAGE="$PROGNAME -d datadir [-c]
+
+where:
+    -d datadir: the chaindata's parent directory
+    -c:         clean up mode
+"
+
+while getopts :hd:c option; do
+  case $option in
+    h)
+      echo "$USAGE"
+      exit 0
+      ;;
+    d) DATADIR=$OPTARG ;;
+    c) CLEANUP=true ;;
+    :)
+      echo "$PROGNAME: option requires an argument -- '$OPTARG'" >&2
+      exit 1
+      ;;
+    *)
+      echo "$PROGNAME: bad option -$OPTARG" >&2
+      echo "Try '$PROGNAME -h' for more information" >&2
+      exit 1
+      ;;
+  esac
+done
+shift $((OPTIND - 1))
+
+if [[ -z $DATADIR ]]; then
+  echo "$PROGNAME: missing mandatory option" >&2
+  echo "Try '$(basename "$0") -h' for more information" >&2
```

Comment:
if [[ -z $DATADIR ]]; then
  error_exit "missing mandatory option. Try '$PROGNAME -h' for more information."
fi